pkg/eventmonitor/eventmonitor.go

```diff
@@ -0,0 +1,262 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package eventmonitor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc"
+
+	"github.com/DataDog/datadog-agent/pkg/eventmonitor/config"
+	"github.com/DataDog/datadog-agent/pkg/security/probe"
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+	"github.com/DataDog/datadog-agent/pkg/security/seclog"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	statsdPoolSize = 64
+)
+
+var (
+	// allowedEventTypes defines allowed event type for subscribers
+	allowedEventTypes = []model.EventType{model.ForkEventType, model.ExecEventType, model.ExitEventType}
+)
+
+// EventMonitor represents the system-probe module for kernel event monitoring
+type EventMonitor struct {
```

Comment:
I wonder if `eventmonitor` should be under `pkg/security` ?
The rational here is that this folder doesn't contain much code, it's mostly just importing / calling code from `pkg/security` anyway (i.e. the logger, probe, the secl model, etc). Unless there is a plan to drastically refactor CWS further, I believe this package should live as part of the security `package`. The rational is mostly because of the config, it feels weird to have the `eventmonitor` package config drastically impact the way `security` probes / ebpf programs behave. Likewise, this would mean that the `security` package now has `yet another place` to import and define its config from.

Reply:
Regardless of where this package lives, can we make sure that the CWS team is the owner of the `eventmonitor` package ? The configuration of this package has a direct impact on how the CWS agent behaves, which has a direct impact on our security coverage and overall product.

Reply:
The event monitor, the security probe and the CWS consumer are very tightly coupled right now. I completely agree that the `eventmonitor` config contains settings that seem outside of its scope, seeing as the only impacted consumer of these settings is the CWS consumer (as mentioned in [your comment below](https://github.com/DataDog/datadog-agent/pull/16108#discussion_r1137962927)). FWIW, the decision on which config values went where was based on necessity rather than logical grouping: all the values that are read by the security probe went in the event monitor config (since it creates the probe), and all the values read by the CWS consumer went into the CWS config. Personally, I think the way forward from here would be to continue to refactor the probe to try and extract as much CWS-specific logic as possible into that consumer (thus allowing us to move the related config settings as well), rather than keep this tight coupling & move the event monitor to the security package - but this is definitely something we can discuss further!

Reply:
Resolving, I think Sylvain's latest changes address the problem 👍🏻